tests/functionTests/sslFunctionTests.txt

```diff
@@ -0,0 +1,145 @@
+| | simple_map_int[0/0]... = {}
+| | simple_map_string[0/0] = {}
+| simple_map_int[0/0]... = {}
+| simple_map_string[0/0] = {}
+RPC succeeded :D
+#END_TEST
+
+##############################################################################
+# Secure Channel with all certificates/keys
+##############################################################################
+
+#START_TEST complete Key/Cert-Options
+@@CMD@@ --complete --ssl --clientCert=@@PTB@@/cert-key-pair/client_crt.pem" "
+-h
+--help
+--ssl
+--clientCert=
+--clientKey=
+--serverCert=
+--complete
+--debugComplete
+--dot
+--noColor
+--color
+--version
+--printParsedMessage
+--noSimpleMapOutput
+--connectTimeoutMilliseconds=
+--customOutput
+unix:
+unix-abstract:
+dns:
+ipv4:
+ipv6:
+#END_TEST
+
+#START_TEST gRPC over Secure Channel 
+@@CMD@@ --ssl --clientCert=@@PTB@@/cert-key-pair/client_crt.pem --clientKey=@@PTB@@/cert-key-pair/client_key.pem --serverCert=@@PTB@@/cert-key-pair/server_crt.pem localhost:50443 examples.NestedTypeRpcs echoNestedMaps
+Warning: no Fields found in parseTree for message 'NestedMaps'
+/.* Received message:
+| number_and_string..... = {NumberAndString}
+| | number = 0 (0x00000000)
+| | str... = ""
+| sub_message........... = {SubMessage}
+| | simple_map_int[0/0]... = {}
+| | simple_map_string[0/0] = {}
+| simple_map_int[0/0]... = {}
+| simple_map_string[0/0] = {}
+RPC succeeded :D
+#END_TEST
+
+##############################################################################
+# Secure Channel with empty certificates/keys (if not user-defined)
+##############################################################################
+
+#START_TEST secure gRPC with missing Certificate/Key and server does not require cert-key-pair
+@@CMD@@ --ssl --clientCert=@@PTB@@/cert-key-pair/client_crt.pem --serverCert=@@PTB@@/cert-key-pair/server_crt.pem localhost:50052 examples.NestedTypeRpcs echoNestedMaps
+Warning: no Fields found in parseTree for message 'NestedMaps'
+/.* Received message:
+| number_and_string..... = {NumberAndString}
+| | number = 0 (0x00000000)
+| | str... = ""
+| sub_message........... = {SubMessage}
+| | simple_map_int[0/0]... = {}
+| | simple_map_string[0/0] = {}
+| simple_map_int[0/0]... = {}
+| simple_map_string[0/0] = {}
+RPC succeeded :D
+#END_TEST
+
+#START_TEST secure gRPC with missing Certificate/Key and server requires cert-key-pair
+@@CMD@@ --ssl --clientCert=@@PTB@@/cert-key-pair/client_crt.pem --serverCert=@@PTB@@/cert-key-pair/server_crt.pem localhost:50443 examples.NestedTypeRpcs echoNestedMaps
+Error: Could not establish Channel. Try checking network connection, hostname or SSL credentials.
+Grammar could not be fetched from the server address: 'localhost:50443'
+#END_TEST
+
+# TODO-Test: secure gRPC with serverCert at default location
+# ./gwhisper --ssl localhost:50052 examples.NestedTypeRpcs echoNestedMaps
+
+#START_TEST secure gRPC with missing serverCert at default location
+@@CMD@@ --ssl localhost:50052 examples.NestedTypeRpcs echoNestedMaps
```

Comment:
not 100% sure how this is supposed to work in the code. I think currently this only works by accident, as I cannot find code handling this scenario explicitly (when cert paths arguments are not given)